File: BOJ/[31831] 과민성 대장 증후군.py
N, M = map(int, input().split())

arr = list(map(int, input().split()))

cnt = 0
result = 0
for a in arr:
    if cnt + a >= 0:
        cnt += a
    else:
        cnt = 0
    if cnt >= M:
        result += 1

print(result)

File: BOJ/[31831] 과민성 대장 증후군.js
let input = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "input.txt")
  .toString()
  .trim()
  .split("\n");


const [N, M] = input[0].split(" ").map(Number);
const arr = input[1].split(" ").map(Number);

let cnt = 0;
let result = 0;

for (const a of arr) {
    if (cnt + a >= 0) {
        cnt += a;
    } else {
        cnt = 0;
    }
    if (cnt >= M) {
        result += 1;
    }
}

console.log(result);